draft-ietf-ipsecme-ikev2-multiple-ke-00.xml

```diff
@@ -0,0 +1,1037 @@
+	</address>
+	</author>
+
+    <author fullname="Valery Smyslov" initials="V." surname="Smyslov">
+    <organization>ELVIS-PLUS</organization>
+    <address><postal><street></street>
+    </postal>
+    <email>svan@elvis.ru</email>
+    </address>
+    </author>
+
+    <date/>
+	<workgroup>Internet Engineering Task Force (IETF)</workgroup>
+	<abstract>
+        <t> This document describes how to extend the Internet Key Exchange Protocol
+		Version 2 (IKEv2) to allow multiple key exchanges to take place 
+        while computation of a shared secret during Security Association (SA) setup.
+        The primary application of this feature in IKEv2 is an ability to perform one or more 
+        post-quantum key exchanges in conjunction with the classical (Elliptic Curve) Diffie-Hellman key exchange,
+        so that the resulting shared key is resistant against quantum computer attacks.
+        Another possible application is a possibility to combine several key exchanges 
+        in situations when no single key exchange algorithm is trusted by both parties.
+        </t>
+
+        <t> This document updates RFC7296 by renaming a tranform type 4 from "Diffie-Hellman Group (D-H)"
+        to "Key Exchange Method (KE)" and renaming a field in the Key Exchange Payload from "Diffie-Hellman Group Num"
+        to "Key Exchange Method". It also renames an IANA registry for this transform type 
+        from "Transform Type 4 - Diffie-Hellman Group Transform IDs" to 
+        "Transform Type 4 - Key Exchange Method Transform IDs". These changes generalize 
+        key exchange algorithms that can be used in IKEv2.
+        </t>
+    </abstract>
+	</front>
+
+	<middle>
+	<section title="Introduction" >
+		<section title="Problem Description" ><t>
+			Internet Key Exchange Protocol (IKEv2) as specified in <xref target="RFC7296"/> 
+            uses the Diffie-Hellman (DH) or Elliptic Curve
+			Diffie-Hellman (ECDH) algorithm to establish a shared secret
+			between an initiator and a responder.  The security of the DH and
+			ECDH algorithms relies on the difficulty to solve a discrete logarithm
+			problem in multiplicative and elliptic curve groups respectively when
+			the order of the group parameter is large enough.  While solving such
+			a problem remains difficult with current computing power, it is
+			believed that general purpose quantum computers will be able to solve
+			this problem, implying that the security of IKEv2 is compromised.
+			There are, however, a number of cryptosystems that are conjectured to
+			be resistant against quantum computer attack.  This family of
+			cryptosystems are known as post-quantum cryptography (PQC).  It is
+			sometimes also referred to as quantum-safe cryptography (QSC) or
+			quantum-resistant cryptography (QRC).
+		</t></section>
+
+		<section title="Proposed Extension" >
+            <t>
+			This document describes a framework to perform multiple successive key 
```

Comment:
```suggestion
			This document describes a method to perform multiple successive key 
```

Reply:
Given that we are dropping the work `framework` from the title, I thought it is not necessary to use this word in the body. This could be just me, but the work `framework` implies something of a high-level or theoretical, whereas in this draft, pretty details description is given on how to do multiple key-exchanges. However, I'm happy to use this word if others agree to leave it in.